dogs/src/app/breed-page/breed-page.component.ts

```diff
@@ -0,0 +1,49 @@
+import { Component, OnInit } from "@angular/core";
+import { ActivatedRoute, Router } from "@angular/router";
+
+import { DogService } from "../shared/api-dog.service";
+
+@Component({
+	selector: "app-breed-page",
+	templateUrl: "./breed-page.component.html",
+	styleUrls: ["./breed-page.component.css"]
+})
+export class BreedPageComponent implements OnInit {
+
+	public breedName: string = "";
+	public imageURL: any = "";
+	public subBreedNames: string[] = [];
+
+	public constructor(
+		private dogService: DogService,
+		private activedRoute: ActivatedRoute,
+		private router: Router
+	) { }
+
+	public ngOnInit(): void {
+		this.prepareBreedNameFromURL();
+		this.prepareSubBreed();
+	}
+
+	private prepareBreedNameFromURL(): void {
+		this.activedRoute.paramMap.subscribe(params => {
+			this.breedName = String(params.get("breedName"));
+			this.dogService.getBreedImage(this.breedName).subscribe(
+				(data) => {
+					this.imageURL = data.message;
+				},
+				(error) => {
+					this.router.navigate(["not-found"]);
```

Comment:
do not navigate to not-found
if your server isn't responding you will get an error. In this case, we would be redirected to not-found page which is wrong. A better approach is to show an error message. For example, toast notifications.